Updated XRM.Navigation.navigateTo to be up to date

KnethM opened the pull request:
Fixes #182

KnethM committed:
Updated pageinput to individual objects

Changed object PageInput to individual objects EntityList, EntityRecord, WebResource.

KnethM committed:
Update XRM.Naviation.navigateTo to be up to date

- added pageinput interface
- added interface extensions
- updated attributes

KnethM committed:
fixed asterix alignment issue

sudhip-git commented:
Could you please confirm when this pull request will be merged

magesoe commented:
I am back from vacation next week, will look at this PR then

magesoe commented on src/XrmDefinitelyTyped/Resources/Extensions/xrm_ext_9-.d.ts:
```diff
@@ -499,7 +560,7 @@ declare namespace Xrm {
          * @param navigationOptions Options for navigating to a page: whether to open inline or in a dialog. If you don't specify this parameter, page is opened inline by default.
          */
-        navigateTo(pageInput: PageInput, navigationOptions: NavigationOptions): Promise<undefined>;
+        navigateTo(pageInput: EntityRecord | EntityList | WebResource, navigationOptions: NavigationOptions): Promise<undefined>;
```
According to the documentation, navigationOptions is optional, so it should be navigationOptions?

magesoe commented on src/XrmDefinitelyTyped/Resources/Extensions/xrm_ext_9-.d.ts:
```diff
@@ -411,22 +411,19 @@ declare namespace Xrm {
     }
 
-    type PageType = "entitylist" | "webresource";
+    type PageType = "entitylist" | "entityrecord" | "webresource";
```
Love the idea you had of extending a common interface, however it does pose a problem. Right now there is no connection between the chosen PageType and the type of extension. This means you are allowed to pass false parameters as this
```typescript
Xrm.Navigation.navigateTo({
      pageType: "entitylist",
      entityName: "",
      createFromEntity: {
        entityType: "",
        id: "",
        name: ""
      }
    })
```
The `pageType` variable should be moved to each declaration with a single allowed string that matches that declaration. For example `interface EntityList` should look like this 
```typescript
interface EntityList {
    pageType: "entitylist"
    ...
}
```
The reason why the base interface is removed is that pageType was the only common type among these interfaces. You should still keep the navigateTo function the same, with the union of the three interfaces.

magesoe reviewed (changes requested):
Love the PR, some minor changes. Do you agree with them?

KnethM committed:
Updates for change request

- removed type PageType
- removed pageinput interface
- added pageType variable to each individual interface
- changed navigationOptions to be optional

KnethM committed:
Linter errors
